central/notifications/datastore/datastore_test.go

```diff
@@ -0,0 +1,168 @@
+package datastore
+
+import (
+	"context"
+	"testing"
+
+	"github.com/pkg/errors"
+	searchMocks "github.com/stackrox/rox/central/notifications/datastore/internal/search/mocks"
+	storeMocks "github.com/stackrox/rox/central/notifications/datastore/internal/store/mocks"
+	writerMocks "github.com/stackrox/rox/central/notifications/datastore/internal/writer/mocks"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/mock/gomock"
+)
+
+var (
+	errFake   = errors.New("fake error")
+	fakeQuery = &v1.Query{}
+)
+
+func TestNotificationsDatastore(t *testing.T) {
+	t.Parallel()
+	suite.Run(t, new(datastoreTestSuite))
+}
+
+type datastoreTestSuite struct {
+	suite.Suite
+	mockCtrl *gomock.Controller
+
+	ctx       context.Context
+	datastore DataStore
+	searcher  *searchMocks.MockSearcher
+	store     *storeMocks.MockStore
+	writer    *writerMocks.MockWriter
+}
+
+func (s *datastoreTestSuite) SetupTest() {
+	s.mockCtrl = gomock.NewController(s.T())
```

Comment:
Technically, you don't need to store the mock controller and call `Finish()` on it since you pass the `testing.T` to it ([see reference here](https://pkg.go.dev/github.com/golang/mock/gomock#Controller.Finish)).

This way you can skip the `TearDownTest()` completely.